packages/fuels-core/src/types/wrappers/message_proof.rs

```diff
@@ -35,35 +46,18 @@ pub struct MessageProof {
 
 impl From<ClientMessageProof> for MessageProof {
     fn from(client_message_proof: ClientMessageProof) -> Self {
-        let message_proof_set = client_message_proof
-            .message_proof
-            .proof_set
-            .iter()
-            .map(|proof| proof.0 .0)
-            .collect();
-        let block_proof_set = client_message_proof
-            .block_proof
-            .proof_set
-            .iter()
-            .map(|proof| proof.0 .0)
-            .collect();
-
+        let sender: Address = client_message_proof.sender;
+        let recipient: Address = client_message_proof.recipient;
         Self {
-            message_proof: MerkleProof {
-                proof_set: message_proof_set,
-                proof_index: client_message_proof.message_proof.proof_index.0,
-            },
-            block_proof: MerkleProof {
-                proof_set: block_proof_set,
-                proof_index: client_message_proof.block_proof.proof_index.0,
-            },
+            message_proof: client_message_proof.message_proof.into(),
+            block_proof: client_message_proof.block_proof.into(),
             message_block_header: client_message_proof.message_block_header.into(),
             commit_block_header: client_message_proof.commit_block_header.into(),
-            sender: client_message_proof.sender.0 .0.into(),
-            recipient: client_message_proof.recipient.0 .0.into(),
-            nonce: client_message_proof.nonce.0 .0,
-            amount: client_message_proof.amount.0,
-            data: client_message_proof.data.0 .0,
+            sender: Bech32Address::from(sender),
+            recipient: Bech32Address::from(recipient),
```

Comment:
```suggestion
            sender: Bech32Address::from(client_message_proof.sender),
            recipient: Bech32Address::from(client_message_proof.recipient),
```

Reply:
Or even shorter:
`client_message_proof.sender.into()` and so on